pkg/proxy/service_test.go

```diff
@@ -0,0 +1,534 @@
+/*
```

Comment:
General feedback about this file: The arguments you pass in for your test helper function have meaningless names most of the time (i.e "somewhere", "blah"). Can you make it so that the name of the arguments makes it clear what it is? My comments below have some examples. Please try to be consistent with how you name things as well. (i.e if you pass in "test-ns" for a namespace, do it everywhere.

Reply:
Done. PTAL, THANKS!